src/GitHub.Exports.Reactive/Collections/TrackingCollection.cs

```diff
@@ -0,0 +1,889 @@
+#if !DISABLE_REACTIVEUI
+using ReactiveUI;
+#else
+using System.Windows.Threading;
+using System.Threading;
+#endif
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using System.Reactive;
+using System.Reactive.Concurrency;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+
+namespace GitHub.Collections
+{
+    /// <summary>
+    /// TrackingCollection is a specialization of ObservableCollection that gets items from
+    /// an observable sequence and updates its contents in such a way that two updates to
+    /// the same object (as defined by an Equals call) will result in one object on
+    /// the list being updated (as opposed to having two different instances of the object
+    /// added to the list).
+    /// It is always sorted, either via the supplied comparer or using the default comparer
+    /// for T
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
```

Comment:
Might be good to document that this class is not thread safe. So it's up to the consumer to synchronize access to it. For example, if you call `RemoveItem` on two threads, you could run into a race condition because access to `sortedIndexCache` is not synchronized.

Is that going to be a problem? Should we consider using a `ConcurrentDictionary`? Or will it be fine in normal usage patterns?